tajo-catalog/tajo-catalog-common/src/main/java/org/apache/tajo/catalog/CatalogService.java

```diff
@@ -172,7 +172,15 @@ PartitionDescProto getPartition(String databaseName, String tableName, String pa
       throws UndefinedPartitionException, UndefinedPartitionMethodException, UndefinedDatabaseException,
       UndefinedTableException;
 
-  List<PartitionDescProto> getPartitions(String databaseName, String tableName);
+  List<PartitionDescProto> getAllPartitions(String databaseName, String tableName) throws UndefinedDatabaseException,
+    UndefinedTableException, UndefinedPartitionMethodException, PartitionNotFoundException;
+
+  List<PartitionDescProto> getPartitionsByAlgebra(PartitionsByAlgebraProto request) throws
+    UndefinedDatabaseException, UndefinedTableException, UndefinedPartitionMethodException,
+    PartitionNotFoundException, UnsupportedException;
+
+  List<PartitionDescProto> getPartitionsByFilter(PartitionsByFilterProto request) throws
```

Comment:
IMO, this method must not throw PartitionNotFoundException, but return an empty list.